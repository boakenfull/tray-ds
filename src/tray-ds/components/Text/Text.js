import React from "react";
import { string, oneOf, bool } from "prop-types";

import { TextStyled } from "./Text.styled";

const Text = ({ children, size, isSecondary }) => (
  <TextStyled size={size} isSecondary={isSecondary}>{children}</TextStyled>
);

Text.propTypes = {
  children: string.isRequired,
  size: oneOf(["xsmall", "small", "default"]),
  isSecondary: bool,
};

Text.defaultProps = {
  size: "default",
  isSecondary: false,
};

export default Text;
