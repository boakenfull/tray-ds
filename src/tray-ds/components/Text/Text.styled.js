import styled, { css } from "styled-components";

import { waterloo, violet } from "../../styles/colourPalette";
import { xsmallFontSize, smallFontSize, defaultFontSize } from "../../styles/variables";

export const TextStyled = styled.p`
  font-weight: normal;
  margin: 0;
  color: ${(props) => (props.isSecondary ? waterloo : violet)};
  ${(props) => props.size === "xsmall" && css`
    font-size: ${xsmallFontSize};
    font-weight: 300;
  `}
  ${(props) => props.size === "small" && `font-size: ${smallFontSize};`}
  ${(props) => props.size === "default" && css`
    font-size: ${defaultFontSize};
    font-weight: 600;
  `}
`;
