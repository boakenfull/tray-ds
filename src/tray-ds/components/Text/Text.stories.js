import React from "react";
import { text, select } from "@storybook/addon-knobs";

import Text from ".";

const sizeOptions = {
  xSmall: "xsmall",
  Small: "small",
  Default: "default",
};

export default { title: "Text" };

export const Default = () => <Text size={select("Size", sizeOptions, "default")}>{text("Text", "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.")}</Text>;
