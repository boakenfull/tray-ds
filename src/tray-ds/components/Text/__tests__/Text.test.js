import React from "react";
import { render } from "@testing-library/react";

import Text from "../";

test("should render the text component", () => {
  const { container } = render(<Text>Text</Text>);
  expect(container).toMatchSnapshot();
})
