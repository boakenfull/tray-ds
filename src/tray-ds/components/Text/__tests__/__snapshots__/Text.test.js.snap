// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the text component 1`] = `
<div>
  <p
    class="sc-AxjAm QFqrR"
  >
    Text
  </p>
</div>
`;
